### DISCUSSION TITLE
GLOBAL模式的默认选择是DIRECT，如何修改其默认配置？Allow customization of GLOBAL有何用处？

### DISCUSSION THREAD

**Discussion by jelly21fish** [MetaCubeX/mihomo#1306] [Q&A]
请教一下开发者和各位大佬们。目前，开启全局模式，默认选择的是DIRECT，虽可以直接修改，但非常反直觉、不合理。手动添加并配置GLOBAL策略组，可以解决这个问题。
但是我发现，如果这么做，策略组的选择列表也随GLOBAL策略组的内容改变了！本来是按proxy-groups的内容排列，现在就只包含了GLOBAL里的策略组，按其顺序排列。这让我百思不得其解。
而且，选择meta内核（尚未合并此修改：Allow customization of GLOBAL https://github.com/MetaCubeX/mihomo/commit/846bdfa8129b9b984a62f04204b4a6925a016834 ）与选择meta alpha内核，结果并无区别，那么这里的修改到底发挥了什么作用？

**Comment by Winson-030** [comment]
看看你的配置文件，这部分可以在配置文件里写的

**Reply by jelly21fish** [reply]
感谢回复。你说的是什么部分？我之前知道的就是，在proxy-groups可以添加组名为GLOBAL的策略组：
` - {name: GLOBAL, type: select, proxies: [节点选择, 自动选择, 手动选择, 全球直连]}`
如此设置之后，全局模式下的策略组选择确实如上所示。但我的问题在于：规则模式下，整个”代理“页面，也就是进行策略组选择的整个页面，变成了仅剩GLOBAL策略组中的这些选项。而在此之前，我还有不同地区、服务的策略组，而且排序也是按proxy-groups本身的顺序排的。
只要删除GLOBAL策略组，一切恢复正常。而你只要在配置文件添加上述GLOBAL策略组，相信可以复现问题。不知道你是否配置过GLOBAL策略组？
当然，我可以把所有策略组都按原来的顺序加到GLOBAL里。但我认为，这并不是设置GLOBAL策略组的预期结果。它的设置不应该改变规则模式下的整个“代理“页面的选项。而alpha内核的修改也没有发挥预期的作用。

**Reply by jelly21fish** [reply]
更正一下：这个问题在不同客户端表现不同。

在Clash Nyanpasu，“代理组”页面只剩下GLOBAL中的这些组。而在OpenClash、Clash Verge Rev，其他策略组还在，但是顺序被打乱了。GLOBAL策略组中的选项，被整体排在了后面。其他不包含在GLOBAL中的策略组被挪到了前面，按照字符顺序排列。

**Comment by jelly21fish** [comment]
类似的疑问也有其他用户反馈过：
https://github.com/MetaCubeX/mihomo/issues/1220
https://github.com/MetaCubeX/mihomo/issues/1246
